src/domain/occurrence/enrolmentTable/EnrolmentTable.tsx

```diff
@@ -0,0 +1,178 @@
+    () => enrolments.every((e) => selectedEnrolments.includes(e.id)),
+    [enrolments, selectedEnrolments]
+  );
+
+  const selectAll = () => {
+    setSelectedEnrolments(enrolments.map((e) => e.id));
+  };
+
+  const unselectAll = () => {
+    setSelectedEnrolments([]);
+  };
+
+  const handleCheckboxChange = (row: EnrolmentFieldsFragment) => {
+    setSelectedEnrolments(
+      selectedEnrolments.includes(row.id)
+        ? selectedEnrolments.filter((e) => e !== row.id)
+        : [...selectedEnrolments, row.id]
+    );
+  };
+
+  const approvedCount = enrolments
+    .filter((e) => e.status === EnrolmentStatus.Approved)
+    .reduce((acc, cur) => acc + cur.studyGroup.groupSize, 0);
+
+  const pendingCount = enrolments
+    .filter((e) => e.status === EnrolmentStatus.Pending)
+    .reduce((acc, cur) => acc + cur.studyGroup.groupSize, 0);
+
+  const columns = [
+    {
+      Header: (
+        <Checkbox
+          id={`${id}_select-all_checkbox`}
+          checked={isAllSelected}
+          onChange={isAllSelected ? unselectAll : selectAll}
+        />
+      ),
+      accessor: (row: EnrolmentFieldsFragment) => (
+        <Checkbox
+          id={`${id}_${row.id}_checkbox`}
+          checked={selectedEnrolments.includes(row.id)}
+          onChange={() => handleCheckboxChange(row)}
+        />
+      ),
+      id: 'selectRow',
+    },
+    {
+      Header: t('occurrenceDetails.enrolmentTable.columnEnrolmentTime'),
+      accessor: (row: EnrolmentFieldsFragment) =>
+        formatDate(new Date(row.enrolmentTime)),
+      id: 'enrolmentTime',
+    },
+    {
+      Header: t('occurrenceDetails.enrolmentTable.columnPersonName'),
+      accessor: (row: EnrolmentFieldsFragment) => row.person?.name,
+      id: 'personName',
+    },
+    {
+      Header: t('occurrenceDetails.enrolmentTable.columnStudyGroupName'),
+      accessor: (row: EnrolmentFieldsFragment) => row.studyGroup.name,
+      id: 'studyGroupName',
+    },
+    {
+      Header: t('occurrenceDetails.enrolmentTable.columnStudyGroupGroupName'),
+      accessor: (row: EnrolmentFieldsFragment) => row.studyGroup.groupName,
+      id: 'studyGroupGroupName',
+    },
+    {
+      Header: t('occurrenceDetails.enrolmentTable.columnGroupSize'),
+      accessor: (row: EnrolmentFieldsFragment) => (
+        <>
+          {row.studyGroup.groupSize} / {row.studyGroup.amountOfAdult}
+        </>
+      ),
+      id: 'groupSize',
+    },
+    {
+      Header: t('occurrenceDetails.enrolmentTable.columnStatus'),
+      accessor: (row: EnrolmentFieldsFragment) => (
+        <EnrolmentStatusBadge status={row.status} />
+      ),
+      id: 'status',
+    },
+    {
+      Header: t('occurrenceDetails.enrolmentTable.columnAdditionalInfo'),
+      accessor: (row: EnrolmentFieldsFragment) => row,
+      Cell: ({
+        row,
+      }: {
+        value: EnrolmentFieldsFragment;
+        row: {
+          getToggleRowExpandedProps: () => void;
+          isExpanded: boolean;
+        };
+      }) => {
```

Comment:
I know that this was taken from my code :) but what do you think, should we make a new type to make this more readable?

Reply:
Tried to make more generic type for the column cell